.github/workflows/python_tests.yml

```diff
@@ -48,24 +33,16 @@ jobs:
       - name: Build solidity contracts
         run: forge build
 
-      - name: Run tests with src
-        if: ${{ needs.paths-filter.outputs.src == 'true' }}
+      - name: Load pytest cache
+        uses: actions/cache@v4
+        with:
+          path: cairo/.pytest_cache
+          key: ${{ runner.os }}-pytest-cache
+
+      - name: Run tests with caching
         env:
           PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION: python
         run: |
           cd cairo
           uv run compile_os
-          uv run pytest -n logical --junitxml=junit.xml -o junit_family=legacy --durations=0 -v
-
-      - name: Run tests without src
-        if: ${{ needs.paths-filter.outputs.src == 'false' }}
-        run: |
-          cd cairo
-          uv run compile_os
-          uv run pytest -n logical --ignore=tests/src --ignore=tests/programs --junitxml=junit.xml -o junit_family=legacy --durations=0 -v
-
-      - uses: codecov/codecov-action@v5.0.5
-        with:
-          directory: ./cairo/coverage/
-      - uses: codecov/test-results-action@v1
-        if: ${{ !cancelled() }}
+          uv run pytest -n logical --durations=0 -v
```

Comment:
nit: for nightly tests, there is "tests" specified 
`uv run pytest "tests" -n logical --durations=0 --no-skip-cached-tests`